vmngclient/session.py

```diff
@@ -19,9 +21,73 @@
 logger = logging.getLogger(get_logger_name(__name__))
 
 
+class SessionNotCreatedError(Exception):
+    pass
+
+
+def assert_admin_role(session: Session, expected_admin: bool) -> None:
+    role = cast(dict, session.get_json("/dataservice/admin/user/role"))
+    if not role["isAdmin"] is expected_admin:
+        raise SessionNotCreatedError("Got different role than expected")
+
+
+def create_session(
+    url: str,
+    port: int,
+    username: str,
+    password: str,
+    admin: bool = True,
+    subdomain: str = None,
+    timeout: int = 30,
+) -> Session:
+    """Factory function that creates session object based on provided arguments.
+
+    Args:
+        url (str): IP address or domain name
+        port (int): port
+        username (str): username
+        password (str): password
+        admin (bool): specifies the type of session that is returned,
+            for 'True' it is 'ProviderSession', if 'False' it is 'TenantSession'
```

Comment:
Can we just use enum SessionType.Provider or SessionType.Tenant ?

create_session( url, 1234, "admin", "admin", **SessionType.Provider**, "xyz")
)